src/test/ui/const-generics/defaults/doesnt_infer.stderr

```diff
@@ -1,10 +1,13 @@
-error[E0282]: type annotations needed for `Foo<{_: u32}>`
-  --> $DIR/doesnt_infer.rs:11:15
+error[E0282]: type annotations needed for `Foo<N>`
+  --> $DIR/doesnt_infer.rs:11:9
    |
 LL |     let foo = Foo::foo();
-   |         ---   ^^^^^^^^ cannot infer the value of const parameter `N`
-   |         |
-   |         consider giving `foo` the explicit type `Foo<N>`, where the const parameter `N` is specified
+   |         ^^^
+   |
+help: consider giving `foo` a type
+   |
+LL |     let foo: Foo<N> = Foo::foo();
+   |            ++++++++
```

Comment:
I think we'll need two different pprinters here, one for the messages and one for the suggestions.

Here, we would want the message to talk about `Foo<{N: u32}>` and the suggestion about `Foo<{ value }>` (ideally `Foo<0i32>`, but whatever, not this PR).